scripts/compile-igs.js

```diff
@@ -0,0 +1,158 @@
+const util = require('util');
+const path = require('path');
+const { readdir, readFile, existsSync } = require('fs');
+
+const readFilePmd = util.promisify(readFile);
+const readDirPmd = util.promisify(readdir);
+
+const compile = require('fhir-works-on-aws-search-es/...?');
+
+async function loadJson(fileName) {
+    return JSON.parse(await readFilePmd(fileName, 'utf8'));
+}
+
+/**
+ *  Helper class used for compiling IGs. Its main functions are:
+ *  - Looks through a folder of IG packs
+ *  - Make sure no dependencies are missing
+ *  - and there are no circular dependencies
+ *  - calls compile function for each SearchParameters
+ *
+ */
+class IGsCompileHelper {
+    constructor(options = {}) {
+        this.options = options;
+    }
+
+    async listIgDirs(parentDir) {
+        return (await readDirPmd(parentDir, { withFileTypes: true }))
+            .filter(dirent => dirent.isDirectory())
+            .map(dirent => path.join(parentDir.toString(), dirent.name));
+    }
+
+    async compileIg(igDir) {
+        const indexJson = path.join(igDir.toString(), '.index.json');
+        if (!existsSync(indexJson)) {
+            return;
+        }
+        const index = await loadJson(indexJson);
+        index.files.forEach(file => {
+            if (file.resourceType === 'SearchParameter') {
+                const filePath = path.join(igDir.toString(), file.filename);
+                console.log(`Compiling ${filePath}`);
+                loadJson(filePath).then(jsonData => {
+                    compile(jsonData);
+                });
+            }
+        });
```

Comment:
We need to rearrange this a little bit. `compile` should be called only once with an array of all the search params from all the IGs that must be compiled. My apologies if this wasn't clear before.

Also, the `compile` output must be saved under `./compiledImplementationGuides` as a json file.

Reply:
Just one JSON file? Should the file be `./compiledImplementationGuides.json` or a JSON file in the `./compiledImplementationGuides` folder. If the latter then what filename should it be? 
`./compiledImplementationGuides/searchParameters.json` maybe, since we are just compiling the `SearchParameter` resources for now. 
Should then different type of resources compiled by different compiles and stored in different files?

Reply:
`./compiledImplementationGuides/searchParameters.json` would work. We need a different file per compiler. Actually, something like `./compiledImplementationGuides/fhir-works-on-aws-search-es.json` is best. Naming it after the module that compiled it is better since future compilers may consume several resource types at once.

> Should then different type of resources compiled by different compiles and stored in different files?

yes, that's the plan. But we are starting with only search for now

Reply:
Maybe the ImplementationGuides interface could have a getName() method. That way the compile-igs script just needs to pass a list of ImplementationGuides to the IGCompiler and the compiler can use the getName() method to get the file name like so: `` const fileName = `${igs.getName()}.json` ``

Reply:
That's a good idea. Although I'm not making additional changes to the interface for the search-only release. I think that we could further explore this idea once we move to the next phase and start working on additional compilers.